haskell/experimental/private/module.bzl

```diff
@@ -619,6 +676,7 @@ def build_haskell_modules(
         # called in all cases to validate cross_library_deps, although the output
         # might be ignored when disabling narrowing
         narrowed_interfaces = _collect_narrowed_deps_module_files(ctx.label, per_module_maps.transitive_interfaces, dep)
+        narrowed_abis = [per_module_maps.abis[m] for m in per_module_maps.abis]
```

Comment:
Revise this because it looks like it collects all the narrowed abis, while we only want the ones from modules that are directly imported.